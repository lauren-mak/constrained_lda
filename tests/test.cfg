[default]
input = /Users/laurenmak/Dropbox/workspace/constrained_lda/tests/test_doc_word.csv
outdir = /Users/laurenmak/Dropbox/workspace/constrained_lda/tests/out

[hyper_params]
num_topics = 3
alpha = 0.1
eta = 0.01
kappa = 0.75

[runtime]
update_iters = 5
gp_iters = 10
gp_thresh = 0.001
