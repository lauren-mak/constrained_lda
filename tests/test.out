00:00:59: Initializing constrained LDA model
00:00:59: Doc_word_df [[1. 0. 1. 0.]
 [1. 0. 0. 1.]
 [0. 1. 1. 0.]]
00:00:59: Lambda [[1.08551306 0.99973866 1.09316002 0.93308883]
 [0.98483888 1.04948334 0.88181034 1.08332674]
 [0.99790379 0.94553684 0.99183328 1.17827033]]
00:00:59: Elogbeta [[-1.73199339 -1.86492231 -1.72087914 -1.98028501]
 [-1.85839903 -1.75460118 -2.04640113 -1.70387918]
 [-1.86850796 -1.95839292 -1.87856902 -1.60477305]]
00:00:59: expElogbeta [[0.17693137 0.15490824 0.17890879 0.13802989]
 [0.15592206 0.17297621 0.12919904 0.18197624]
 [0.15435379 0.14108497 0.15280861 0.20093515]]
00:00:59: Iteration 0
00:00:59: Document 0
00:00:59: Initial gamma 0 [0.80713297 0.95899369 0.94502127]
00:00:59: 0 Gamma 0 [0.72132814 0.77122465 0.80744721]
00:00:59: 0 deltaGamma 0 0.13704931026424558
00:00:59: 1 Gamma 0 [0.7720853  0.71027087 0.81764382]
00:00:59: 1 deltaGamma 0 0.040635847733621056
00:00:59: 2 Gamma 0 [0.85747714 0.61583031 0.82669255]
00:00:59: 2 deltaGamma 0 0.06296037626112294
00:00:59: 3 Gamma 0 [0.99757449 0.47685767 0.82556784]
00:00:59: 3 deltaGamma 0 0.09339823848968747
00:00:59: 4 Gamma 0 [1.21038065 0.29812215 0.7914972 ]
00:00:59: 4 deltaGamma 0 0.1418707696150916
00:00:59: 5 Gamma 0 [1.47362993 0.14269104 0.68367904]
00:00:59: 5 deltaGamma 0 0.17549951889769214
00:00:59: 6 Gamma 0 [1.71183546 0.10080252 0.48736201]
00:00:59: 6 deltaGamma 0 0.15880369162100022
00:00:59: 7 Gamma 0 [1.93242678 0.10003806 0.26753516]
00:00:59: 7 deltaGamma 0 0.14706087494865838
00:00:59: 8 Gamma 0 [2.07732773 0.1000324  0.12263987]
00:00:59: 8 deltaGamma 0 0.09660063591386207
00:00:59: 9 Gamma 0 [2.09976076 0.10002983 0.10020941]
00:00:59: 9 deltaGamma 0 0.0149553545969326
00:00:59: Document 1
00:00:59: Initial gamma 1 [1.06467945 0.96092276 1.09279099]
00:00:59: 0 Gamma 1 [0.75205994 0.687613   0.86032706]
00:00:59: 0 deltaGamma 1 0.272797732626288
00:00:59: 1 Gamma 1 [0.70198257 0.63776662 0.96025081]
00:00:59: 1 deltaGamma 1 0.06661583381458953
00:00:59: 2 Gamma 1 [0.62077005 0.55458178 1.12464816]
00:00:59: 2 deltaGamma 1 0.10959823731078418
00:00:59: 3 Gamma 1 [0.49382596 0.42438465 1.38178939]
00:00:59: 3 deltaGamma 1 0.17142748075876654
00:00:59: 4 Gamma 1 [0.31827512 0.25406234 1.72766255]
00:00:59: 4 deltaGamma 1 0.23058210879474905
00:00:59: 5 Gamma 1 [0.15208942 0.12288293 2.02502764]
00:00:59: 5 deltaGamma 1 0.19824339750278022
00:00:59: 6 Gamma 1 [0.10115915 0.10024393 2.09859692]
00:00:59: 6 deltaGamma 1 0.04904618545533914
00:00:59: 7 Gamma 1 [0.1000377  0.10003595 2.09992636]
00:00:59: 7 deltaGamma 1 0.000886289123872272
00:00:59: Document 2
00:00:59: Initial gamma 2 [1.12368765 1.05707757 1.01695129]
00:00:59: 0 Gamma 2 [0.87787436 0.73829    0.68383564]
00:00:59: 0 deltaGamma 2 0.2992388354453731
00:00:59: 1 Gamma 2 [1.00505113 0.69616836 0.59878051]
00:00:59: 1 deltaGamma 2 0.08478450824868244
00:00:59: 2 Gamma 2 [1.21128083 0.61998314 0.46873603]
00:00:59: 2 deltaGamma 2 0.13748647293912744
00:00:59: 3 Gamma 2 [1.51452575 0.48982539 0.29564886]
00:00:59: 3 deltaGamma 2 0.20216327693647504
00:00:59: 4 Gamma 2 [1.85436213 0.3036425  0.14199537]
00:00:59: 4 deltaGamma 2 0.22655758693054207
00:00:59: 5 Gamma 2 [2.05830118 0.14094099 0.10075782]
00:00:59: 5 deltaGamma 2 0.13595936914229287
00:00:59: 6 Gamma 2 [2.09929727 0.100667   0.10003573]
00:00:59: 6 deltaGamma 2 0.027330724852162844
00:00:59: 7 Gamma 2 [2.0999316  0.10003599 0.10003241]
00:00:59: 7 deltaGamma 2 0.0004228872352895763
00:00:59: Sstats [[9.99988576e-01 9.99962787e-01 1.99994186e+00 1.26132182e-05]
 [3.47145044e-05 2.04993659e-05 2.65079877e-05 1.66261032e-05]
 [9.99976709e-01 1.67138866e-05 3.16337184e-05 9.99970761e-01]]
00:00:59: Lambda [[0.82498062 0.75977348 1.65062396 0.00933972]
 [0.00987403 0.01051097 0.00883563 0.01084678]
 [0.75838945 0.00946722 0.00994186 0.89545962]]
00:00:59: Elogbeta [[  -1.92399836   -2.07635668   -0.84646213 -108.64669024]
 [ -76.3657507   -70.22765725  -88.26972537  -67.28169598]
 [  -1.25183708 -106.37648224 -101.3328524    -0.95080566]]
00:00:59: expElogbeta [[1.46021945e-01 1.25386203e-01 4.28929751e-01 6.53645013e-48]
 [6.83558778e-34 3.16603695e-31 4.62322842e-39 6.02463449e-30]
 [2.85978948e-01 6.32820818e-47 9.81073097e-45 3.86429567e-01]]
00:00:59: E[log p(theta | alpha) - log q(theta | gamma)] = -35.30101086168883
00:00:59: E[log p(beta | eta) - log q (beta | lambda)] = -10.67708497993825
00:00:59: Logl: -45.97809584162708
00:00:59: Iteration 1
00:00:59: Document 0
00:00:59: Initial gamma 0 [0.93165001 0.85132879 1.08020252]
00:00:59: 0 Gamma 0 [1.38591809 0.1        0.81408191]
00:00:59: 0 deltaGamma 0 0.49057248978385354
00:00:59: 1 Gamma 0 [1.64611261 0.1        0.55388739]
00:00:59: 1 deltaGamma 0 0.17346301143644968
00:00:59: 2 Gamma 0 [1.87073485 0.1        0.32926515]
00:00:59: 2 deltaGamma 0 0.14974816477130268
00:00:59: 3 Gamma 0 [2.04465622 0.1        0.15534378]
00:00:59: 3 deltaGamma 0 0.11594757597015597
00:00:59: 4 Gamma 0 [2.09859108 0.1        0.10140892]
00:00:59: 4 deltaGamma 0 0.03595657626647424
00:00:59: 5 Gamma 0 [2.09995872 0.1        0.10004128]
00:00:59: 5 deltaGamma 0 0.0009117602063075705
00:00:59: Document 1
00:00:59: Initial gamma 1 [1.29280227 0.90795977 0.90723773]
00:00:59: 0 Gamma 1 [0.57291354 0.1        1.62708646]
00:00:59: 0 deltaGamma 1 0.7492324148679609
00:00:59: 1 Gamma 1 [0.1784109 0.1       2.0215891]
00:00:59: 1 deltaGamma 1 0.26300175594200165
00:00:59: 2 Gamma 1 [0.10088365 0.1        2.09911635]
00:00:59: 2 deltaGamma 1 0.05168483439110227
00:00:59: 3 Gamma 1 [0.10001021 0.1        2.09998979]
00:00:59: 3 deltaGamma 1 0.0005822912200494912
00:00:59: Document 2
00:00:59: Initial gamma 2 [0.84843657 0.98021743 1.02426473]
00:00:59: 0 Gamma 2 [2.1 0.1 0.1]
00:00:59: 0 deltaGamma 2 1.018681865722359
00:00:59: 1 Gamma 2 [2.1 0.1 0.1]
00:00:59: 1 deltaGamma 2 1.4802973661668753e-16
00:00:59: Sstats [[9.99973377e-01 1.00000000e+00 2.00000000e+00 3.09717795e-52]
 [1.29344131e-37 4.61857757e-35 3.94308948e-43 2.85170806e-34]
 [1.00002662e+00 9.23151588e-51 8.38498796e-49 1.00000000e+00]]
00:00:59: Lambda [[6.26968802e-01 5.77427844e-01 2.49244218e+00 9.33971527e-05]
 [9.87402997e-05 1.05109687e-04 8.83563469e-05 1.08467761e-04]
 [5.76391124e-01 9.46722113e-05 9.94186433e-05 6.80549308e-01]]
00:00:59: Elogbeta [[-2.61223161e+00 -2.79153067e+00 -4.66758121e-01 -1.07087078e+04]
 [-7.63178361e+03 -7.01807749e+03 -8.82201259e+03 -6.72353574e+03]
 [-1.41040595e+00 -1.05631197e+04 -1.00588337e+04 -1.05745289e+00]]
00:00:59: expElogbeta [[0.07337063 0.06132727 0.62703174 0.        ]
 [0.         0.         0.         0.        ]
 [0.24404419 0.         0.         0.3473394 ]]
00:00:59: E[log p(theta | alpha) - log q(theta | gamma)] = -37.35401562857187
00:00:59: E[log p(beta | eta) - log q (beta | lambda)] = -591.7491607968465
00:00:59: Logl: -629.1031764254184
00:00:59: Iteration 2
00:00:59: Document 0
00:00:59: Initial gamma 0 [0.89292957 1.06797228 1.08596609]
00:00:59: 0 Gamma 0 [1.27852524 0.1        0.92147476]
00:00:59: 0 deltaGamma 0 0.5060197542012478
00:00:59: 1 Gamma 0 [1.43597316 0.1        0.76402684]
00:00:59: 1 deltaGamma 0 0.10496527933805351
00:00:59: 2 Gamma 0 [1.55605567 0.1        0.64394433]
00:00:59: 2 deltaGamma 0 0.08005500642591752
00:00:59: 3 Gamma 0 [1.66847349 0.1        0.53152651]
00:00:59: 3 deltaGamma 0 0.07494521522269611
00:00:59: 4 Gamma 0 [1.78938094 0.1        0.41061906]
00:00:59: 4 deltaGamma 0 0.08060497085021112
00:00:59: 5 Gamma 0 [1.92652677 0.1        0.27347323]
00:00:59: 5 deltaGamma 0 0.09143055297232612
00:00:59: 6 Gamma 0 [2.05385936 0.1        0.14614064]
00:00:59: 6 deltaGamma 0 0.08488839040912648
00:00:59: 7 Gamma 0 [2.09843355 0.1        0.10156645]
00:00:59: 7 deltaGamma 0 0.029716127938671715
00:00:59: 8 Gamma 0 [2.09992879 0.1        0.10007121]
00:00:59: 8 deltaGamma 0 0.000996827656662568
00:00:59: Document 1
00:00:59: Initial gamma 1 [1.02776286 0.87767431 0.83123184]
00:00:59: 0 Gamma 1 [0.40174543 0.1        1.79825457]
00:00:59: 0 deltaGamma 1 0.790238153765573
00:00:59: 1 Gamma 1 [0.11739089 0.1        2.08260911]
00:00:59: 1 deltaGamma 1 0.189569694933857
00:00:59: 2 Gamma 1 [0.10002506 0.1        2.09997494]
00:00:59: 2 deltaGamma 1 0.01157722164560481
00:00:59: 3 Gamma 1 [0.10000551 0.1        2.09999449]
00:00:59: 3 deltaGamma 1 1.3031279072132454e-05
00:00:59: Document 2
00:00:59: Initial gamma 2 [1.06808626 0.87755694 0.97173486]
00:00:59: 0 Gamma 2 [2.1 0.1 0.1]
00:00:59: 0 deltaGamma 2 0.893735179230649
00:00:59: 1 Gamma 2 [2.1 0.1 0.1]
00:00:59: 1 deltaGamma 2 1.4802973661668753e-16
00:00:59: Sstats [[0.99994422 1.         2.         0.        ]
 [0.         0.         0.         0.        ]
 [1.00005578 0.         0.         1.        ]]
00:00:59: Lambda [[4.76470061e-01 4.38845162e-01 3.76358769e+00 9.33971527e-07]
 [9.87402997e-07 1.05109687e-06 8.83563469e-07 1.08467761e-06]
 [4.38081366e-01 9.46722113e-07 9.94186433e-07 5.17217474e-01]]
00:00:59: Elogbeta [[-3.51691787e+00 -3.73340309e+00 -2.45733979e-01 -1.07069846e+06]
 [-7.63178311e+05 -7.01807700e+05 -8.82201208e+05 -6.72353526e+05]
 [-1.65247375e+00 -1.05627609e+06 -1.00584749e+06 -1.22771378e+00]]
00:00:59: expElogbeta [[0.02969081 0.02391132 0.78213026 0.        ]
 [0.         0.         0.         0.        ]
 [0.19157541 0.         0.         0.29296159]]
00:00:59: E[log p(theta | alpha) - log q(theta | gamma)] = -40.60592825433241
00:00:59: E[log p(beta | eta) - log q (beta | lambda)] = -61471.98590516244
00:00:59: Logl: -61512.59183341677
00:00:59: Iteration 3
00:00:59: Document 0
00:00:59: Initial gamma 0 [0.99220115 1.01534461 0.95417453]
00:00:59: 0 Gamma 0 [1.24193225 0.1        0.95806775]
00:00:59: 0 deltaGamma 0 0.3896563088979725
00:00:59: 1 Gamma 0 [1.28950763 0.1        0.91049237]
00:00:59: 1 deltaGamma 0 0.0317169173012295
00:00:59: 2 Gamma 0 [1.31237423 0.1        0.88762577]
00:00:59: 2 deltaGamma 0 0.015244401970492483
00:00:59: 3 Gamma 0 [1.32430975 0.1        0.87569025]
00:00:59: 3 deltaGamma 0 0.007957016744719972
00:00:59: 4 Gamma 0 [1.33080406 0.1        0.86919594]
00:00:59: 4 deltaGamma 0 0.004329536815185146
00:00:59: 5 Gamma 0 [1.33441711 0.1        0.86558289]
00:00:59: 5 deltaGamma 0 0.002408702340177834
00:00:59: 6 Gamma 0 [1.33645197 0.1        0.86354803]
00:00:59: 6 deltaGamma 0 0.001356574386432996
00:00:59: 7 Gamma 0 [1.33760589 0.1        0.86239411]
00:00:59: 7 deltaGamma 0 0.000769279485762849
00:00:59: Document 1
00:00:59: Initial gamma 1 [0.94467684 0.96537768 1.01932486]
00:00:59: 0 Gamma 1 [0.22018636 0.1        1.97981364]
00:00:59: 0 deltaGamma 1 0.8501189791975113
00:00:59: 1 Gamma 1 [0.1008415 0.1       2.0991585]
00:00:59: 1 deltaGamma 1 0.07956324291355703
00:00:59: 2 Gamma 1 [0.10000309 0.1        2.09999691]
00:00:59: 2 deltaGamma 1 0.0005589397943856304
00:00:59: Document 2
00:00:59: Initial gamma 2 [0.89987658 1.11365181 0.81368163]
00:00:59: 0 Gamma 2 [2.1 0.1 0.1]
00:00:59: 0 deltaGamma 2 0.9758189555444283
00:00:59: 1 Gamma 2 [2.1 0.1 0.1]
00:00:59: 1 deltaGamma 2 1.4802973661668753e-16
00:00:59: Sstats [[0.23826563 1.         2.         0.        ]
 [0.         0.         0.         0.        ]
 [1.76173437 0.         0.         1.        ]]
00:00:59: Lambda [[8.99095305e-02 3.33522323e-01 5.68301741e+00 9.33971527e-09]
 [9.87402997e-09 1.05109687e-08 8.83563469e-09 1.08467761e-08]
 [5.83218063e-01 9.46722113e-09 9.94186433e-09 3.93085280e-01]]
00:00:59: Elogbeta [[-1.32858405e+01 -4.85536273e+00 -7.83074053e-02 -1.07069648e+08]
 [-7.63178311e+07 -7.01807700e+07 -8.82201208e+07 -6.72353526e+07]
 [-9.86198390e-01 -1.05627616e+08 -1.00584756e+08 -1.99559077e+00]]
00:00:59: expElogbeta [[1.69837184e-06 7.78650841e-03 9.24680131e-01 0.00000000e+00]
 [0.00000000e+00 0.00000000e+00 0.00000000e+00 0.00000000e+00]
 [3.72991969e-01 0.00000000e+00 0.00000000e+00 1.35933325e-01]]
00:00:59: E[log p(theta | alpha) - log q(theta | gamma)] = -40.639078836093105
00:00:59: E[log p(beta | eta) - log q (beta | lambda)] = -6152280.235292579
00:00:59: Logl: -6152320.874371415
00:00:59: Iteration 4
00:00:59: Document 0
00:00:59: Initial gamma 0 [0.98032275 0.98315198 0.88916177]
00:00:59: 0 Gamma 0 [1.10000538 0.1        1.09999462]
00:00:59: 0 deltaGamma 0 0.4045558187828233
00:00:59: 1 Gamma 0 [1.10000455 0.1        1.09999545]
00:00:59: 1 deltaGamma 0 5.477223099514816e-07
00:00:59: Document 1
00:00:59: Initial gamma 1 [0.9516482  1.05889052 0.84517507]
00:00:59: 0 Gamma 1 [0.10000559 0.1        2.09999441]
00:00:59: 0 deltaGamma 1 1.0217841548252442
00:00:59: 1 Gamma 1 [0.1 0.1 2.1]
00:00:59: 1 deltaGamma 1 3.728922421420792e-06
00:00:59: Document 2
00:00:59: Initial gamma 2 [1.02422805 0.95344141 0.96207176]
00:00:59: 0 Gamma 2 [2.1 0.1 0.1]
00:00:59: 0 deltaGamma 2 0.9304283741215983
00:00:59: 1 Gamma 2 [2.1 0.1 0.1]
00:00:59: 1 deltaGamma 2 1.4802973661668753e-16
00:00:59: Sstats [[4.55349573e-06 1.00000000e+00 2.00000000e+00 0.00000000e+00]
 [0.00000000e+00 0.00000000e+00 0.00000000e+00 0.00000000e+00]
 [1.99999545e+00 0.00000000e+00 0.00000000e+00 1.00000000e+00]]
00:00:59: Lambda [[8.99402357e-04 2.53476965e-01 8.58135629e+00 9.33971527e-11]
 [9.87402997e-11 1.05109687e-10 8.83563469e-11 1.08467761e-10]
 [8.80657284e-01 9.46722113e-11 9.94186433e-11 2.98744813e-01]]
00:00:59: Elogbeta [[-1.11454632e+03 -6.28957969e+00 -3.09535499e-02 -1.07069645e+10]
 [-7.63178311e+09 -7.01807700e+09 -8.82201208e+09 -6.72353526e+09]
 [-4.77261492e-01 -1.05627616e+10 -1.00584756e+10 -3.20249043e+00]]
00:00:59: expElogbeta [[0.         0.00185554 0.96952061 0.        ]
 [0.         0.         0.         0.        ]
 [0.62048026 0.         0.         0.04066081]]
00:00:59: E[log p(theta | alpha) - log q(theta | gamma)] = -41.608678716172456
00:00:59: E[log p(beta | eta) - log q (beta | lambda)] = -615235990.1998997
00:00:59: Logl: -615236031.8085784
00:00:59: Finished inferring strains and composition, printing results
